reclaim.c

```diff
@@ -158,7 +158,7 @@ GC_INLINE word *GC_clear_block(word *p, word sz, signed_word *count)
       p += 2;
     }
 # else
-    p += sizeof(word *); /* Skip link field */
+    p = (ptr_t)p + sizeof(word *); /* Skip link field */
```

Comment:
Given that `p` is of type `word *`, is this different than `*p++`? I guess it depends if `sizeof(word *)` always `==` `sizeof(word)` ?

Reply:
In CHERI systems, `sizeof(word *) -> (16) != sizeof(word) -> (8)` .  This was why the freelist was getting clobbered. The increment `p++` was only incrementing by `sizeof(word) ->(8)` while we need it to skip over by `sizeof(word *) -> (16)` or any type of pointer.

Reply:
Got it!